maven-core/src/main/java/org/apache/maven/graph/MultiModuleCollectionStrategy.java

```diff
@@ -0,0 +1,181 @@
+package org.apache.maven.graph;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.building.ModelProblem;
+import org.apache.maven.plugin.PluginManagerException;
+import org.apache.maven.plugin.PluginResolutionException;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.ProjectBuildingException;
+import org.apache.maven.project.ProjectBuildingResult;
+import org.codehaus.plexus.logging.Logger;
+import org.eclipse.aether.resolution.ArtifactResolutionException;
+import org.eclipse.aether.transfer.ArtifactNotFoundException;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+/**
+ * Strategy for collecting Maven projects from the multi-module project root, even when executed in a submodule.
+ */
+@Named( "MultiModuleCollectionStrategy" )
+@Singleton
+public class MultiModuleCollectionStrategy implements ProjectCollectionStrategy
+{
+    private final Logger logger;
+    private final ProjectCollector projectCollector;
+
+    @Inject
+    public MultiModuleCollectionStrategy( Logger logger, ProjectCollector projectCollector )
```

Comment:
No need to inject Loggers anymore, instead simply do `private static final Logger LOGGER = LoggerFactory...`

Reply:
Should that be the SLF4J Logger + Factory?

Reply:
yes